user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/registration/MeasurementUpload.java

```diff
@@ -258,54 +260,76 @@ private void runValidation(ArrayList<ValidationRequest> requests, int itemsToVal
     AtomicInteger counter = new AtomicInteger();
     counter.set(1);
 
+    UI ui = UI.getCurrent();
+    validationStarted(ui);
     service.validate(Flux.fromIterable(requests))
-        .doFirst(this::validationStarted)
         .doOnNext(ignored -> counter.incrementAndGet())
         .bufferTimeout(20, Duration.ofMillis(200))
-        .doOnNext(item -> setValidationProgressText(
+        .doOnNext(item -> setValidationProgressText(ui,
             "Processed " + counter.get() + " requests from " + itemsToValidate))
         .flatMap(Flux::fromIterable)
         .collectList()
-        .doFinally(it -> validationFinished())
-        .subscribe(responses -> displayValidationResults(
-            responses, itemsToValidate, fileName));
+        .doOnSuccess(it -> validationFinished(ui))
+        .subscribe(responses -> ui.access(() -> {
+          validationFinished(ui);
+          displayValidationResults(responses, itemsToValidate, fileName);
+          ui.push(); // Enforce server push
+        }));
   }
 
-  private void validationStarted() {
-    showValidationProgress();
-    setValidationProgressText("Starting validation ...");
+  private void validationStarted(UI ui) {
+    if (ui == null || !ui.isAttached()) {
+      return;
+    }
+    showValidationProgress(ui);
+    setValidationProgressText(ui, "Starting validation ...");
     isValidationInProgress.set(true);
   }
 
-  private void validationFinished() {
-    hideValidationProgress();
+  private void validationFinished(UI ui) {
+    if (ui == null || !ui.isAttached()) {
+      return;
+    }
+    hideValidationProgress(ui);
     isValidationInProgress.set(false);
   }
 
 
-  private void hideValidationProgress() {
-    getUI().ifPresent(ui -> ui.access(() -> validationProgress.setVisible(false)));
+  private void hideValidationProgress(@NonNull UI ui) {
+    if (ui.isAttached() && ui.getSession().hasLock()) {
+      validationProgress.setVisible(false);
+    } else {
+      ui.access(() -> validationProgress.setVisible(false));
+    }
   }
 
-  private void showValidationProgress() {
-    getUI().ifPresent(ui -> ui.access(() -> validationProgress.setVisible(true)));
+  private void showValidationProgress(@NonNull UI ui) {
+    if (ui.isAttached() && ui.getSession().hasLock()) {
+      validationProgress.setVisible(true);
+    } else {
+      ui.access(() -> validationProgress.setVisible(true));
+    }
```

Comment:
Nice but is `validationProgess` already populated with the correct information here?